src/openms/include/OpenMS/QC/Identifications.h

```diff
@@ -0,0 +1,87 @@
+// ETH Zurich, and Freie Universitaet Berlin 2002-2020.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Axel Walter $
+// $Authors: Axel Walter $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/QC/QCBase.h>
+
+/**
+ * @brief Detected Proteins/Peptides as a Proteomics QC metric
+ *
+ * Simple class to return the number of detected proteins/peptides
+ * from a given idXML file.
+ *
+ */
+
+namespace OpenMS
+{
+  class OPENMS_DLLAPI Identifications : public QCBase
+  {
+  public:
+    /// Constructor
+    Identifications() = default;
+
+    /// Destructor
+    virtual ~Identifications() = default;
+
+    // stores DetectedCompounds values calculated by compute function
+    struct OPENMS_DLLAPI Result
+    {
+      UInt peptide_spectrum_matches = 0;
+      std::tuple<UInt, float> unique_peptides = {0, 0.0};
+      std::tuple<UInt, float> unique_proteins = {0, 0.0};
+      float missed_cleavages_mean = 0;
+      double protein_hit_scores_mean = 0;
+      double peptide_length_mean = 0;
+
+      bool operator==(const Result& rhs) const;
+    };
+
+     /**
+    @brief computes the number of detected compounds in a featureXML file
+
+    @param inputfile_id idXML file 
+    @return ??
+
+    **/
+    Result compute(const String& inputfile_id);
```

Comment:
same here. Use a structure. Maybe even a template, so people can pass in featureMaps or consensusMaps as well?